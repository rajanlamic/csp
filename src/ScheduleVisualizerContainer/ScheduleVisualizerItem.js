import React from 'react';
import { Grid } from 'semantic-ui-react';

export const ScheduleVisualizerItem = ({item}) => {
    return (<Grid.Row>
          <Grid.Column width={4}>{item.startTime}</Grid.Column>
          <Grid.Column width={4}>{item.endTime}</Grid.Column>
          <Grid.Column width={8}>{item.titleName}</Grid.Column>
        </Grid.Row>)
}
