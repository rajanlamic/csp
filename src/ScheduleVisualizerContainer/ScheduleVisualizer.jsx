import React from 'react';
import { ScheduleVisualizerItem } from './ScheduleVisualizerItem';
import { Grid } from 'semantic-ui-react';

const ScheduleVisualizer = (props) => {
    return (<Grid divided="vertically" padded="horizontally">
      {
        props.schedules && props.schedules.length && props.schedules.map((item, indx) => {
          return <ScheduleVisualizerItem key={indx} item={item} />;
        })
      }
      {
        !props.schedules && ('No schedules to be displayed')
      }
      
    </Grid>)
}

export default ScheduleVisualizer;
