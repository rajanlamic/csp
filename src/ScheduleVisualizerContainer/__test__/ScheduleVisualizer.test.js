import * as React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import ScheduleVisualizer from '../ScheduleVisualizer';
import { ScheduleVisualizerItem } from '../ScheduleVisualizerItem';

describe('ScheduleVisualizer', () => {
  let component, props;
  
  beforeEach(() => {
    props = {
      schedules: [
        {
          startTime: "06:00",
          endTime: "07:00",
          titleName: "The Amazing Spiderman"
        }
      ]
    }
    component = shallow(<ScheduleVisualizer />);
  })
  afterEach(() => {
    component = null;
    props = null;
  })
  
  it('should render ScheduleVisualizer component properly using snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render ScheduleVisualizerItem if schedule is found', () => {
    component = shallow(<ScheduleVisualizer {...props} />);
    expect(component.find(ScheduleVisualizerItem).length).toEqual(1);
  })

  it('should not render ScheduleVisualizerItem if no schedule is found', () => {
    component = shallow(<ScheduleVisualizer />);
    expect(component.find(ScheduleVisualizerItem).length).toEqual(0);
  })
  
})
