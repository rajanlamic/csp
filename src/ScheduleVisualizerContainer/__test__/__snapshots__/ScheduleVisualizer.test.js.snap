// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleVisualizer should render ScheduleVisualizer component properly using snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScheduleVisualizer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        undefined,
        "No schedules to be displayed",
      ],
      "divided": "vertically",
      "padded": "horizontally",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      "No schedules to be displayed",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          undefined,
          "No schedules to be displayed",
        ],
        "divided": "vertically",
        "padded": "horizontally",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        "No schedules to be displayed",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
